### PR TITLE
Fix #3897 - Incorrect behavior of NetflowHeaderV9 Count

### LINKED ISSUE #3897: NetflowHeaderV9 Counts flowsets instead of records
### Brief description

According to the [RFC 3954](https://www.rfc-editor.org/rfc/rfc3954#section-5.1), the count field in the Netflow header is defined as :

>    Count
         The total number of records in the Export Packet, which is the
         sum of Options FlowSet records, Template FlowSet records, and
         Data FlowSet records.

However, the [NetflowHeaderV9](https://github.com/secdev/scapy/blob/master/scapy/layers/netflow.py#L1261-L1269) counts the number of flowsets within the Netflow packet

```python
# scapy.layers.netflow.NetflowHeaderV9
# ...
    def post_build(self, pkt, pay):
        if self.count is None:
            count = sum(1 for x in self.layers() if x in [
                NetflowFlowsetV9,
                NetflowDataflowsetV9,
                NetflowOptionsFlowsetV9]
            )
            pkt = struct.pack("!H", count) + pkt[2:]
        return pkt + pay
```
Can you confirm it is a bug ? 

### Scapy version

2.4.5

### Python version

3.10.6

### Operating system

Linux 5.15.0

### Additional environment information

_No response_

### How to reproduce

```python
from scapy.all import IP, UDP, Ether, wrpcap
from scapy.fields import IntField, XByteField
from scapy.layers.inet import UDP
from scapy.layers.netflow import (GetNetflowRecordV9, NetflowDataflowsetV9,
                                  NetflowFlowsetV9, NetflowHeader,
                                  NetflowHeaderV9,
                                  NetflowOptionsFlowsetOptionV9,
                                  NetflowOptionsFlowsetScopeV9,
                                  NetflowOptionsFlowsetV9, NetflowRecordV9,
                                  NetflowTemplateFieldV9, NetflowTemplateV9)
from scapy.packet import Packet

# Let's define two templates, 256 for IPv4 and 257 for IPv6
template_flowset = NetflowFlowsetV9(
    templates=[ NetflowTemplateV9(
        template_fields=[
            NetflowTemplateFieldV9(fieldType="IN_BYTES", fieldLength=1),
            NetflowTemplateFieldV9(fieldType="IN_PKTS", fieldLength=4),
            NetflowTemplateFieldV9(fieldType="PROTOCOL"),
            NetflowTemplateFieldV9(fieldType="IPV4_SRC_ADDR"),
            NetflowTemplateFieldV9(fieldType="IPV4_DST_ADDR"),
        ],
        templateID=256,
        fieldCount=5),
        NetflowTemplateV9(
        template_fields=[
            NetflowTemplateFieldV9(fieldType="IN_BYTES", fieldLength=1),
            NetflowTemplateFieldV9(fieldType="IN_PKTS", fieldLength=4),
            NetflowTemplateFieldV9(fieldType="PROTOCOL"),
            NetflowTemplateFieldV9(fieldType="IPV6_SRC_ADDR"),
            NetflowTemplateFieldV9(fieldType="IPV6_DST_ADDR"),
        ],
        templateID=257,
        fieldCount=5)
    ],
    flowSetID=0
)

# Generate classes for data records
Record256 = GetNetflowRecordV9(template_flowset, templateID = 256)
Record257 = GetNetflowRecordV9(template_flowset, templateID = 257)

# Now lets build a dataFlowSet with 5* #256 records
dataFlowset_1 = NetflowDataflowsetV9(
    templateID=256,
    records=[
        Record256(
            IN_BYTES=b"\x12",
            IN_PKTS=b"\0\0\0\0",
            PROTOCOL=1,
            IPV4_SRC_ADDR="192.168.0.10",
            IPV4_DST_ADDR="192.168.0.11"
        ),
        Record256(
            IN_BYTES=b"\x0c",
            IN_PKTS=b"\1\1\1\1",
            PROTOCOL=2,
            IPV4_SRC_ADDR="172.0.0.10",
            IPV4_DST_ADDR="172.0.0.11"
        ),
        Record256(
            IN_BYTES=b"\x0c",
            IN_PKTS=b"\1\1\1\1",
            PROTOCOL=3,
            IPV4_SRC_ADDR="172.0.0.10",
            IPV4_DST_ADDR="172.0.0.11"
        ),
        Record256(
            IN_BYTES=b"\x0c",
            IN_PKTS=b"\1\1\1\1",
            PROTOCOL=4,
            IPV4_SRC_ADDR="172.0.0.10",
            IPV4_DST_ADDR="172.0.0.11"
        ),
        Record256(
            IN_BYTES=b"\x0c",
            IN_PKTS=b"\1\1\1\1",
            PROTOCOL=5,
            IPV4_SRC_ADDR="172.0.0.10",
            IPV4_DST_ADDR="172.0.0.11"
        )
    ],
)

# and another one of 2*257 Records
dataFlowset_2 = NetflowDataflowsetV9(
    templateID=257,
    records=[
        Record257(
            IN_BYTES=b"\x12",
            IN_PKTS=b"\0\0\0\0",
            PROTOCOL=1,
            IPV6_SRC_ADDR="2001:db8:3333:4444:5555:6666:7777:8888",
            IPV6_DST_ADDR="2001:db8::"
        ),
        Record257(
            IN_BYTES=b"\x0c",
            IN_PKTS=b"\1\1\1\1",
            PROTOCOL=2,
            IPV6_SRC_ADDR="2001:db8:3333:4444:CCCC:DDDD:EEEE:FFFF",
            IPV6_DST_ADDR="2001:db8::"
        )
    ],
)

# An option template flowset, containing an unique template
opttmpl258_flowSet = NetflowOptionsFlowsetV9(
    templateID = 258,
    option_scope_length = 4*1,
    option_field_length = 4*2,
    scopes = [
        NetflowOptionsFlowsetScopeV9(scopeFieldType= 1,	scopeFieldlength= 4),
    ],
    options = [
        NetflowOptionsFlowsetOptionV9(optionFieldType= 34,	optionFieldlength= 4),
        NetflowOptionsFlowsetOptionV9(optionFieldType= 35,	optionFieldlength= 1)
    ])

# And finally a Record class for #258 Options
class Record_258(NetflowRecordV9):
    name = "Option interface-table"
    fields_desc = [
        IntField("System", 0),
        IntField("SAMPLING_INTERVAL", 4),
        XByteField("SAMPLING_ALGORITHM", 1)
    ]
    match_subclass = True

# to generate a last record Flowset
optiondataFlowset = NetflowDataflowsetV9(
    templateID=258,
    records=[
        Record_258(
            System=424242,
            SAMPLING_INTERVAL=100,
            SAMPLING_ALGORITHM=0x01
        ),
        Record_258(
            System=242424,
            SAMPLING_INTERVAL=1000,
            SAMPLING_ALGORITHM=0x02
        )
    ],
)

netflow_header = NetflowHeader() / NetflowHeaderV9(unixSecs=1547927349.328283)
pkt =  netflow_header / template_flowset / opttmpl258_flowSet / dataFlowset_1 / dataFlowset_2 / optiondataFlowset
#      Count: 12      =        2        +         1           +       5       +       2       +        2

pkt.show2()
wrpcap('bad_netflow_header_count.pcap', Ether() / IP() / UDP() / pkt)

# Count = 5, the number of flowsets and not the total number of records,
# Sould be 12
```

### Actual result

```
###[ Netflow Header ]### 
  version   = 9
###[ Netflow Header V9 ]### 
     count     = 5
     ...
```

### Expected result

```
###[ Netflow Header ]### 
  version   = 9
###[ Netflow Header V9 ]### 
     count     = 12
     ...
```

### Related resources

[bad_netflow_header_count.zip](https://github.com/secdev/scapy/files/10724793/bad_netflow_header_count.zip)

### PR DESCRIPTION
**Pull request by theplatypus**
Fixes #3897

Tested successfully in local. 

About `NetflowOptionsFlowsetV9` count hardcoded to `1`, it comes from the fact that [NetflowOptionsFlowsetV9.templateID](https://github.com/secdev/scapy/blob/master/scapy/layers/netflow.py#L1640) not being a `PacketListField`, hence it can only carry a single template, but the correction would be outside of the scope of this issue.

I tried my best to conform with scapy's coding style, please tell me how to correct things, I will gladly correct it following your guidelines.

### CODE REVIEW

**Comment by gpotter2** [comment]
Hi ! could you please rebase against master? Thanks !

**Comment by theplatypus** [comment]
Hi,
Rebase done, and add a comment about the remaining flake8 issue on https://github.com/secdev/scapy/pull/3898/commits/38ed682f9b0301975dcfcf10f4abd1e6a69c2b28#r1105538293

File: scapy/layers/netflow.py [hunk 1]
```diff
@@ -1259,12 +1260,21 @@ class NetflowHeaderV9(Packet):
...
+
         if self.count is None:
-            count = sum(1 for x in self.layers() if x in [
-                NetflowFlowsetV9,
-                NetflowDataflowsetV9,
-                NetflowOptionsFlowsetV9]
-            )
+            count = sum(sum(count_by_layer(self.getlayer(layer_cls, nth))
+                            for nth in range(1, n + 1))
+                            for layer_cls, n in Counter(self.layers()).items())
```
Comment (gpotter2) [inline scapy/layers/netflow.py:1277]: You could do
```suggestion
            count = sum(
                sum(count_by_layer(self.getlayer(layer_cls, nth)) for nth in range(1, n + 1))
                for layer_cls, n in Counter(self.layers()).items()
            )
```

**Review by gpotter2 — COMMENTED** [review]
Sorry for the delay reviewing !

File: scapy/layers/netflow.py (hunk 1 above)
Comment (theplatypus) [reply scapy/layers/netflow.py]: Fine for me :+1:

**Comment by theplatypus** [comment]
Mmmmh, I do not see how the unit tests are failing for pypy3.9, on a test set not involved with my changes ?
Moreover I don't think it failed before, any idea ?  :thinking:

**Comment by gpotter2** [comment]
You can ignore that safely. There's still a flake8 error though :/

**Comment by theplatypus** [comment]
Fixed so that flake8 does not raise an error anymore

**Comment by theplatypus** [comment]
Hi, (:
What do you think of the last modification (only pypy3.9 related tests failed) ?
Thanks

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**